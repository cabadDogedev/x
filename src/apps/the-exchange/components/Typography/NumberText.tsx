import { ReactNode } from 'react';

type NumberTextProps = {
  children: ReactNode;
  className?: string;
}

const NumberText = ({ children, className }: NumberTextProps) => {
  return (
    <h1 className={`text-black_grey ${className}`}>{children}</h1>
  )
}

export default NumberText;
