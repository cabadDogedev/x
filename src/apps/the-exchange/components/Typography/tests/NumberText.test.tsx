import renderer, { ReactTestRendererJSON } from 'react-test-renderer';

// components
import NumberText from '../NumberText';

describe('<NumberText />', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <>
          <NumberText>Some number text.</NumberText>
          <NumberText className='text-red-500'>Some red heading red text</NumberText>
          <NumberText className='text-[120px]'>Some heading with font size 120px</NumberText>
        </>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();

    const treeElements = tree as ReactTestRendererJSON[];

    expect(treeElements[0].children?.length).toBe(1);
    expect(treeElements[0].children?.[0]).toBe('Some number text.');
    expect(treeElements[0].type).toBe('h1');
    expect(treeElements[0].props.className).toContain('text-black_grey');
    expect(treeElements[1].props.className).toContain('text-red-500');
    expect(treeElements[2].props.className).toContain('text-[120px]');
  });
});

