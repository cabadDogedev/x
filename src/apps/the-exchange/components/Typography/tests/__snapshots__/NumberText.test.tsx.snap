// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NumberText /> renders correctly 1`] = `
[
  <h1
    className="text-black_grey undefined"
  >
    Some number text.
  </h1>,
  <h1
    className="text-black_grey text-red-500"
  >
    Some red heading red text
  </h1>,
  <h1
    className="text-black_grey text-[120px]"
  >
    Some heading with font size 120px
  </h1>,
]
`;
